wx/tools/img2py.py

```diff
@@ -152,7 +148,9 @@ def img2py(image_file, python_file,
         app = wx.App(0)
 
     # convert the image file to a temporary file
-    tfname = tempfile.mktemp()
+    with tempfile.NamedTemporaryFile(delete=False) as fid:
+        tfname = fid.name
+
     try:
```

Comment:
If you indent this block inside the `with` block, then you can also remove the deleting of the `tfname` file in the `finally` block and let `NamedTemporaryFile` take care of the cleanup.

Reply:
Well, it is not as simple as that.  NamedTemporaryFile() creates a file for writing and keeps it open until the "with block" ends. If I move the part inside the "try-finally block" inside the "with block" the fact that the file is still open will block the convert function to write to that file with the result that nothing will be converted.
For your idea to work the convert function must be refactored to also allow the fid to be passed in as an argument to convert function instead of the tfname string so that it can write to it directly (without opening it again since it is already open for writing).
However, the convert function is not so easy to refactor for the case when the image file is not the right type and you have to the wx.img2img conversion before saving.

So what do think?

Reply:
Okay, it's alright the way it is then. I just thought it looked odd to still have to explicitly close the file and didn't think it through all the way.

I'll merge this in a little while.